File: python/api-eventbridge-lambda/lambda/event_producer_lambda.py
import json
import boto3
import datetime
import logging

logger = logging.getLogger()
logger.setLevel(logging.INFO)


def lambda_handler(event, context):
    eventbridge_client = boto3.client('events')
    request_body = event["body"]
    if request_body is None:
        request_body = ""
    # Structure of EventBridge Event
    eventbridge_event = {
        'Time': datetime.datetime.now(),
        'Source': 'com.mycompany.myapp',
        'Detail': request_body,
        'DetailType': 'service_status'
    }
    logger.info(eventbridge_event)

    # Send event to EventBridge
    response = eventbridge_client.put_events(
        Entries=[
            eventbridge_event
        ]
    )

    logger.info(response)

    # Returns success reponse to API Gateway
    return {
        "statusCode": 200,
        "body": json.dumps({
            "result": "from Producer"
        }),
    }

File: typescript/api-eventbridge-lambda/lambda/eventProducerLambda.ts
const AWS = require('aws-sdk')
const eventBridgeClient = new AWS.EventBridge();

exports.handler = async function(event:{body:any, [key:string]:any}) {
    try {
        console.log('Request:', JSON.stringify(event, undefined, 2));

        let requestBody = event["body"];
        if (!requestBody) {
            requestBody = '';
        }

        // Structure of EventBridge Event
        const eventbridgeEvent = {
            'Time': new Date(),
            'Source': 'com.mycompany.myapp',
            'Detail': requestBody,
            'DetailType': 'service_status'
        }

        console.log("EventBridge event: ", eventbridgeEvent);

        // Send event to EventBridge
        const response = await eventBridgeClient.putEvents({
            Entries: [
                eventbridgeEvent
            ]
        }).promise();

        console.log('EventBridge response:', response);
    } catch (error) {
        console.error("There was a problem: ", error);
    }

    return {
      statusCode: 200,
      headers: { "Content-Type": "text/plain" },
      body: `from Producer...`
    };
  };

